src/include/storage/checkpoint_io.h

```diff
@@ -0,0 +1,500 @@
+#pragma once
```

Comment:
There's a ton of stuff that's happening in this file, many classes. Ideally this should be split apart into multiple files to be maintainable and readable.